pkg/templates/pdbminavailable/template.go

```diff
@@ -103,6 +104,33 @@ func minAvailableCheck(lintCtx lintcontext.LintContext, object lintcontext.Objec
 	for _, dl := range deploymentLikes {
 		pdbMinAvailable := value
 		replicas, _ := extract.Replicas(dl)
+		if int(replicas) == 1 {
+			hpaList, err := getHorizontalPodAutoscalers(lintCtx, pdb.Namespace)
+			if err != nil {
+				break
+			}
+			for _, hpa := range hpaList {
+				// Select from the list of HPAs one that have scaleTargetRef that matches the deployment name
+				hpa, ok := hpa.(*autoscalingV2.HorizontalPodAutoscaler)
+				if !ok {
+					return nil
+				}
+				// Ensure that the hpa scaleTargetRef is the deployment name
+				if hpa.Spec.ScaleTargetRef.Name != dl.GetName() {
+					continue
+				}
+				// If the HPA has a minReplicas set, use that value as the deployment replica, else fail
+				if hpa.Spec.MinReplicas != nil {
+					replicas = *hpa.Spec.MinReplicas
+				} else {
```

Comment:
If we `continue` in the `if` we can remove `else` and reduce indentation.

Reply:
I think I've implemented the corrections stated, but I think this one is an opportunity to deliver a different, possible, error. 

With the else statement, we cover the case where there's no HPA available and the deployment spec.replicas is nil

Though this scenario won't happen right now as the `extract pod_spec Replicas` function currently enforces a return 1, that could change (I've actually tried to change that but its tied to other tests), as it is possible to have nil replicas. 

Not sure what is the best approach here. To YAGNI or to not YAGNI?